tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/IncClassEmitter.scala

```diff
@@ -0,0 +1,62 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js tools             **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2016, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+package org.scalajs.core.tools.linker.backend.emitter
+
+import scala.collection.mutable
+
+import org.scalajs.core.tools.sem._
+
+import org.scalajs.core.tools.linker.backend.OutputMode
+import org.scalajs.core.tools.linker.LinkingUnit
+
+/** Tracks incrementality breaking changes made during class emitting.
+ */
+final class IncClassEmitter(semantics: Semantics, outputMode: OutputMode) {
+
+  private var currentClassesCtorOpt: Set[String] = Set.empty
+  private[this] val askedForCtorOpt =
+    mutable.Map.empty[String, Set[(String, String, Boolean)]]
+
+  def beginRun(unit: LinkingUnit,
+      invalidateFunc: (String, String, Boolean) => Unit): ScalaJSClassEmitter = {
+    val classEmitter = new ScalaJSClassEmitter(outputMode, unit, this)
+    val newClassesCtorOpt = if (outputMode == OutputMode.ECMAScript6StrongMode) {
+      Set.empty[String]
+    } else {
+      classEmitter.candidateForJSConstructorOpt
+    }
+    val classesWhoChanged = (currentClassesCtorOpt -- newClassesCtorOpt) ++
+      (newClassesCtorOpt -- currentClassesCtorOpt)
```

Comment:
"who" is for people. It should be "that". That said, there is [precedent](https://github.com/scala-js/scala-js/blob/master/tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala#L313-L315) to name such a concept `changedClasses` rather than `classesThatChanged`.

Also, style (not allowed to break after `++` if not in `()` or `{}`):
```scala
val changedClasses = {
  (currentClassesCtorOpt -- newClassesCtorOpt) ++
  (newClassesCtorOpt -- currentClassesCtorOpt)
}
```